### ISSUE TITLE
git workshop improvements before FACN2 & FACG2?

### ISSUE THREAD

**Issue by jsms90** [foundersandcoders/master-reference#394]
[@NataliaLKB's "learn git basics" workshop](https://github.com/NataliaLKB/learn-git-basics) was replaced with [@PiotrBerebecki's git workshop for two](https://github.com/foundersandcoders/git-workflow-workshop-for-two) (issue #286)

This was based on feedback from FAC10 about the workshop:
> + Make git workflow workshop more team based. We ended up doing it all on our own, even if we pair programmed it, it would have been working on your own in terms of working on the repo you were working on. Person 1 PR, person 2 merge it. There were problems with people merging to master during projects.

With FAC11, we ended up stopping the workshop half an hour early, to go through [the github flow](https://guides.github.com/introduction/flow/) because there was a lot of confusion. This was partially caused by the [forking](https://github.com/foundersandcoders/git-workflow-workshop-for-two/issues/29) element of the workshop, but there were lots of questions about the flow as well:
+ what a branch is for i.e. a feature in the github flow
+ why you need to add before committing
+ why you need to checkout master, pull down the remote master, checkout your branch, `git merge master` before pushing
+ how much you put in a commit (very granular)

These are all explained in much more detail in @NataliaLKB's workshop, which has been worked on since it was delivered to FAC10. I am also concerned that most of FAC11's understanding of the above concepts has come from verbal explanations from the mentors.

But the pairing element of @PiotrBerebecki's would be a shame to lose. My preference would be to combine the best elements of both. But my key question now is, how do we move forward, before FACN2 & FACG2?

**Comment by eliascodes** [comment]
Is there anyone actually available to make these changes?

**Comment by NataliaLKB** [comment]
> My preference would be to combine the best elements of both

Why not just recommend both?

**Comment by jsms90** [comment]
@eliascodes I will happily make the changes. I've seen the same problems occur for 3 cohorts now.

@NataliaLKB Links to both will definitely stay in the `master-reference`, but each curriculum can only deliver one. So the question is either "which one? who picks? can the campuses diverge?" or "can we make a workshop that takes the best from both?"

**Comment by NataliaLKB** [comment]
I believe mine used to be a prereq and actually that makes more sense. Should we just do that again?

**Comment by jsms90** [comment]
@NataliaLKB Ahhh really? I didn't know it was a pre-req! That could make a lot of sense, especially now that we're asking students to make their own github website. Maybe it could end up going in our Prep Course...? :thinking:

Anyway, no, I definitely wouldn't suggest making a third one!! The last thing I would want to do is dilute things even further. In the short term, it's just about how to make sure that FACN2 & FACG2 learn what they need to about git (and github?) in week 1.

**Comment by NataliaLKB** [comment]
@jsms90 yep! It was a prereq for FAC5 (I think) and FAC6. Maybe for longer as I am not sure when it stopped being a prereq actually.

**Comment by jsms90** [comment]
#### Discussion between @eliascodes @jsms90 @bradreeder (@des-des consulted via gitter)
What to do in the event that we learn something from the delivery in one campus, that might be beneficial to change before the week gets repeated in Nazareth / Gaza:
1. Criteria / bar for what is deemed "urgent"
    + Difficult to set "criteria". Likely to be on a case by case basis, at least for now.
    + To some degree, isn't that what the stand-ups are for?
2. Who decides whether / what changes get made & who responsible for changing it
    + Having different maintainers for each different workshop / morning challenge makes iterative changes during the curriculum possible. But constant change makes it difficult for mentors to know what material they are about to deliver.
    + Should there be a distinction between small-scale changes / medium-scale / large-scale? e.g. [small scale changes to git workshop](https://github.com/foundersandcoders/git-workflow-workshop-for-two/issues/33) were carried out immediately, but should [larger scale changes](https://github.com/foundersandcoders/git-workflow-workshop-for-two/issues/29) wait?
    + Key question: is keeping the current version going to have a significantly negative impact on the week for the coming cohort?
    + If the change is deemed urgent, CF to get in touch with mentors who are about to deliver it - find out how comfortable are they with material. Are you about to mess up a mentor's prep?
3. By what day does it need to be changed for Naz / Gaza so that mentors can be sufficiently prepared
    + Communication with other CFs & mentors of that week / that workshop
    + Attach release versions to workshops / `master-reference` so (worst case scenario) a cohort can just wind back the clock and teach the version they're comfortable with?
